File: backend/src/utils/cors-origins.ts
/** Origens permitidas para CORS (domínio público + extras opcionais). */
export function resolveCorsOrigins(): string[] {
  const origins = new Set<string>(["http://localhost:3000"]);
  const primary = process.env.FRONTEND_URL?.trim().replace(/\/$/, "");
  if (primary) origins.add(primary);
  const extra = process.env.ALLOWED_ORIGINS?.split(",")
    .map((s) => s.trim().replace(/\/$/, ""))
    .filter(Boolean);
  for (const o of extra ?? []) origins.add(o);
  return [...origins];
}

File: ai-agent/app/cors_origins.py
import os


def resolve_cors_origins() -> list[str]:
    origins: set[str] = {"http://localhost:3000"}
    primary = (os.getenv("FRONTEND_URL") or "").strip().rstrip("/")
    if primary:
        origins.add(primary)
    extra = (os.getenv("ALLOWED_ORIGINS") or "").strip()
    if extra:
        for part in extra.split(","):
            v = part.strip().rstrip("/")
            if v:
                origins.add(v)
    return sorted(origins)
